File: src/producer.js
const Pulsar = require('pulsar-client');

(async () => {
  // Token based authentication
  const tokenStr = '{JWT token}';
	const pulsarUri = 'pulsar+ssl://{host}:6651';
	const topicName = 'persistent://{tenant}/{namespace}/{topic}';
  
  const trustStore = '/etc/ssl/certs/ca-bundle.crt';
  // Debian Ubuntu:
  // const trustStore = '/etc/ssl/certs/ca-certificates.crt'
  // OSX:
  // Export the default certificates to a file, then use that file:
  // security find-certificate -a -p /System/Library/Keychains/SystemCACertificates.keychain > ./ca-certificates.crt
  // trust_certs='./ca-certificates.crt'

  const auth = new Pulsar.AuthenticationToken({token: tokenStr});
 
  // Create Pulsar Client
  const client = new Pulsar.Client({
    serviceUrl: pulsarUri,
    authentication: auth,
    tlsTrustCertsFilePath: trustStore,
    operationTimeoutSeconds: 30,
  });

  // Create a producer
  const producer = await client.createProducer({
    topic: topicName,
  });

  // Send messages
  for (let i = 0; i < 10; i += 1) {
    producer.send({
      data: Buffer.from(`nodejs-message-${i}`)
    });
    console.log(`send message ${i}`);
  }
  await producer.flush();

  await producer.close();
  await client.close();
})();

File: cpp/producer.cc
#include <iostream>

#include <pulsar/Client.h>

using namespace pulsar;
using namespace std;

int main() {

    std::string token = "eyJhbGciOiJ Pulsar JWT";
    std::string serviceURL = "pulsar+ssl://useast2.aws.kafkaesque.io:6651";
    std::string topicName = "persistent://ming-luo/local-useast2-aws/test-topic2";

    AuthenticationPtr auth = pulsar::AuthToken::createWithToken(token);
    ClientConfiguration config = ClientConfiguration();
    config.setAuth(auth);
    config.setTlsTrustCertsFilePath("/etc/ssl/certs/ca-bundle.crt");
    /**
     *Use default CA certs for your environment
     * RHEL/CentOS:
     * trust_certs='/etc/ssl/certs/ca-bundle.crt'
     * Debian/Ubuntu:
     * trust_certs='/etc/ssl/certs/ca-certificates.crt'
     */
    Client client(serviceURL, config);

    Producer producer;
    Result result = client.createProducer(topicName, producer);
    if (result != ResultOk) {
        cout << "Error creating producer: " << result << endl;
        return -1;
    }

    // Send synchronously
    Message msg = MessageBuilder().setContent("content").build();
    Result res = producer.send(msg);
    cout << "Message sent: " << res << endl;

    client.close();
}
